$OpenBSD: patch-src_library_parallel_R_detectCores_R,v 1.1 2020/06/08 18:36:28 feinerer Exp $

Use the number of CPUs online instead of configured.
Upstream SVN revision 78662.

Index: src/library/parallel/R/detectCores.R
--- src/library/parallel/R/detectCores.R.orig
+++ src/library/parallel/R/detectCores.R
@@ -36,7 +36,7 @@ detectCores <-
                      darwin = if(logical) "/usr/sbin/sysctl -n hw.logicalcpu 2>/dev/null" else "/usr/sbin/sysctl -n hw.physicalcpu 2>/dev/null",
                      solaris = if(logical) "/usr/sbin/psrinfo -v | grep 'Status of.*processor' | wc -l" else "/bin/kstat -p -m cpu_info | grep :core_id | cut -f2 | uniq | wc -l",
                      freebsd = "/sbin/sysctl -n hw.ncpu 2>/dev/null",
-                     openbsd = "/sbin/sysctl -n hw.ncpu 2>/dev/null",
+                     openbsd = "/sbin/sysctl -n hw.ncpuonline 2>/dev/null",
                      irix  = c("hinv | grep Processors | sed 's: .*::'", "hinv | grep '^Processor '| wc -l"))
             nm <- names(systems)
             m <- pmatch(nm, R.version$os); m <- nm[!is.na(m)]
